src/video_info/player_response/microformat.rs

```diff
@@ -0,0 +1,62 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::video_info::player_response::video_details::Thumbnail;
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct Microformat {
+    pub player_microformat_renderer: PlayerMicroformatRenderer,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct PlayerMicroformatRenderer {
+    pub available_countries: Vec<String>,
+    // TODO: maybe also an enum
+    pub category: String,
+    pub description: SimpleText,
+    pub embed: Embed,
+    pub external_channel_id: String,
+    pub has_ypc_metadate: bool,
+    pub is_unlisted: bool,
+    pub length_seconds: String,
+    pub live_brodcast_details: Option<LiveBroadcastDetails>,
+    pub owner_channel_name: String,
+    pub owner_profile_url: String,
+    #[serde(with = "crate::serde_impl::date")]
+    pub publish_date: DateTime<Utc>,
+    #[serde(rename = "thumbnail")]
+    #[serde(serialize_with = "Thumbnail::serialize_vec")]
+    #[serde(deserialize_with = "Thumbnail::deserialize_vec")]
+    pub thumbnails: Vec<Thumbnail>,
+    pub title: SimpleText,
+    #[serde(with = "crate::serde_impl::date")]
+    pub upload_date: DateTime<Utc>,
+    pub view_count: i32,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct Embed {
+    pub flash_url: String,
+    pub flash_secure_url: String,
+    pub iframe_url: String,
+    pub height: i32,
+    pub width: i32,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct SimpleText {
+    simple_text: String,
+}
```

Comment:
This might be a `playability_status::Reason`. Maybe I'm wrong, it just looks similar.

Reply:
I don't know. I've looked in some jsons and always found them like that.

Reply:
Ok, let's go with that for now. We can still fix it later if there are other variants of that struct.